arrow-instances/src/main/kotlin/arrow/instances/tuple.kt

```diff
@@ -1,9 +1,49 @@
 package arrow.instances
 
-import arrow.typeclasses.Eq
-import arrow.typeclasses.Monoid
+import arrow.*
 import arrow.core.*
-import arrow.instance
+import arrow.typeclasses.*
+
+@instance(Tuple2::class)
+interface Tuple2FunctorInstance : Functor<Tuple2KindPartial<*>> {
```

Comment:
Star projections work well as input types, but not as output ones as users won't be able to work with the first value of the tuple without casting, as it's not carried over these transformations. Check how other HK2 types like Either do it.

Reply:
Whoops, didn't notice that.